app/controllers/internships_controller.rb

```diff
@@ -0,0 +1,60 @@
+class InternshipsController < ApplicationController
+  before_action :set_cohort, only: [:index, :new, :create, :edit, :update, :destroy]
+  before_action :set_internship, only: [:show, :edit, :update, :destroy]
+
+  def index
+    @internships = @cohort.internships
+  end
+
+  def show
+    @company = @internship.company
+  end
+
+  def new
+    @internship = Internship.new
+  end
+
+  def create
+    @internship = @cohort.internships.new(internship_params)
+
+    if @internship.save
+      flash[:notice] = "Internship added"
+      redirect_to cohort_internships_path(@cohort)
+    else
+      render :new
+    end
+  end
+
+  def edit
+  end
```

Comment:
Oh I see. I'm generally not a fan of before actions that make ivars magically appear :) I think better to be explicit here.